packages/cli/internal/pkg/cli/logs_adapter.go

```diff
@@ -35,6 +38,22 @@ func (o *logsAdapterOpts) Validate() error {
 		return err
 	}
 
+	// It's convenient for the tests to be able to not actually have the ctxManager set.
+	if o.ctxManager != nil {
+		ctxMap, err := o.ctxManager.List()
+		if err != nil {
+			return err
+		}
+
+		summary := ctxMap[o.contextName]
+		engine := summary.Engines[0].Engine
+		if engine == constants.TOIL {
```

Comment:
I think we can avoid explaining code using comments if we use self documenting code instead. It will have additional benefit of forward looking design. In this case the better option might be to introduce a list of engines with native WES API support.

```go
var HasNativeWES = map[sting]bool{constants.TOIL: true}
...
if HasNativeWES[engine] {
  return fmt.Errorf("'%s' doesn't use an adapter, because it supports WES API natively", engine)
}



```

Reply:
I've set up a map for this, and plugged it in to the `environment.go` stuff that already needed this information.